friture: init at 0.36

laikq opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
Note: `scipy` was not listed in the requirements of `setup.py` in the repository, but the package won't build without it.

###### Motivation for this change
Theoretically, it should be possible to "install" Friture via an AppImage. However, I could not get this to work, neither with `appimage-run` nor `appimageTools.wrapType2` (`OS Error: PortAudio library not found`). Building this package from source requires a lot of machine resources, but at least it works. There are instructions on reducing the package size in `setup.py`, but I tried to keep the build as simple as possible because I'm a complete newbie to building packages.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Maintainers

I could of course try to maintain the package, but I was hesitant to add myself to the maintainers list.

laikq committed:
friture: init at 0.36

Lassulus commented:
If you want to maintain this, then please add yourself to the maintainers list in a seperate commit. I think a maintainer for this package would be very much appreciated.
Also, can't we build this with python3Packages instead of python35Packages?

laikq commented:
Apparently, when I try to build the package with `python3Packages`, a unittest fails that is trying to run `friture/filter_design.py`. This module tries to `from scipy.misc import factorial`, and this import fails. `factorial` has been moved from `scipy.misc` to `scipy.special` in version 1.3.0, and was [deprecated since version 1.0.0](https://docs.scipy.org/doc/scipy-1.2.1/reference/generated/scipy.misc.factorial.html#scipy.misc.factorial). It seems like `python3Packages` uses Python 3.7 on my machine, which requires SciPy 1.3.0. Using `python35Packages` fixes this problem for now.

laikq commented:
So, fixed one error, another opened: When I try to run Friture, I get a Qt error à la `Could not find the Qt platform plugin "xcb" in ""`. I think I cleaned up some of my system packages, and somewhere there I had `qtbase` installed (and now not anymore).

Lassulus commented:
ok, I tested it, seems to work fine. Can you squash the commits into one?

laikq commented:
Yes, it works for me too :)

Lassulus reviewed (approved):
tested with nix-review, started friture, analyzed my microphone audio
